### PR TITLE
fix: Error in Pricing Page Validation for Signup Now Modal

### LINKED ISSUE #490: [Bug] - Error in Pricing Page Validation for Signup Now Modal
**Description:**

On the pricing page here, https://documenso.com/pricing, when I click **_Signup Now_**, a modal pops up and then when I click Close (X) at the right side top corner, it validates the form instead of closing it immediately. The error validation persists even after closing the modal and seems to go off only when the page is refreshed

Please refer the video below for more reference :)

[screen-capture (15).webm](https://github.com/documenso/documenso/assets/23498248/62ccdbf8-cf0d-49ed-844e-b0909697a6a4)

### PR DESCRIPTION
**Pull request by haaarsh619**
## Issue number #490 

- Resolves #490

### CODE REVIEW

File: apps/marketing/src/components/(marketing)/claim-plan-dialog.tsx [hunk 1]
```diff
@@ -90,7 +90,11 @@ export const ClaimPlanDialog = ({ className, planId, children }: ClaimPlanDialog
       });
     }
   };
-
+  useEffect(() => {
+    if (!isSubmitting && !open) {
+      reset();
+    }
+  }, [open]);
```
Comment (dguyen) [inline apps/marketing/src/components/(marketing)/claim-plan-dialog.tsx:97]: Could you please add some spacing around the useEffect and surrounding code?

**Comment by dguyen** [comment]
@haaarsh619 thanks for handling this!

/tip 30

**Comment by haaarsh619** [comment]
@dguyen thank you for allowing me to contribute and helping me in the process. 
Do i have to click this link ?

**Comment by dguyen** [comment]
@haaarsh619 no worries

>  Do i have to click this link ?

No, I think you should have received it now. Please let us know if there are any issues.

**Pull request merged**